mcnp2cad.cpp

```diff
@@ -799,10 +799,39 @@ entity_collection_t GeometryContext::defineCell(  CellCard& cell,  bool defineEm
       break;
     case CellCard::MBODYFACET:
       {
-        int cellnum = std::abs(token.second) / 10;
-        int facet = std::abs(token.second) - (cellnum*10);
-        std::cerr << "Attempting to define facet " << facet << " of cell " << cellnum << std::endl;
-        throw std::runtime_error( "Macrobody facets are not yet supported by mcnp2cad." );
+        int identifier = -std::abs( token.second );
+        int cellnum = -identifier / 10;
+        int facet = -identifier - ( cellnum * 10 );
+
+        try{
+          makeSurface( deck.lookup_surface_card( identifier ), NULL, facet);
+          SurfaceVolume& surf = makeSurface( deck.lookup_surface_card( identifier ) );
+          const std::string& mnemonic = deck.lookup_surface_card( identifier )->getMnemonic();
+          bool pos = true;
+          if( mnemonic == "rcc" || mnemonic == "rec" ){
```

Comment:
Right.  So perhaps a separate function, and I'd like to discuss this to make sure all the possible cases are covered - it seems like there aren't enough cases to get them all - but I could be wrong.